.github/workflows/ci-docs-autogen.yml

```diff
@@ -0,0 +1,74 @@
+name: Autogenerated documentation
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  autogen-docs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: yarn
+
+      - name: Install Foundry
+        uses: onbjerg/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Node dependencies cache
+        uses: actions/cache@v3
+        with:
+          path: "node_modules"
+          key: yarn-${{ hashFiles('yarn.lock') }}
```

Comment:
JSYK, this is redundant with the action `setup-node` with parameter `cache: yarn`